### DISCUSSION TITLE
adding Files to PreCache List

### DISCUSSION THREAD

**Discussion by Zaphod-Beeblebrox** [quasarframework/quasar#15016] [CLI - PWA mode] [answered]
My PWA needs access to a few TensorFlow AI Models.  I've put them in my /public folder and the app accesses them just fine when online (It's grabbing them from the server I presume).  When Offline, the files aren't part of the PreCache list in service Worker (sw.js), so loading the model fails.  I've tried to add them to the workboxOptions in quasar.config.js:
`workboxOptions: {
        skipWaiting: true,
        include: [
          "group1-shard1of6.bin",
          "group1-shard2of6.bin",
          "group1-shard3of6.bin",
          "group1-shard4of6.bin",
          "group1-shard5of6.bin",
          "group1-shard6of6.bin",
        ],
      },`

But this doesn't seem to work.  How do I add files that aren't specifically referenced in the various components to the PreCache List?

**Comment by Zaphod-Beeblebrox** [comment, marked as answer]
for anyone else who runs into this issue.  I eventually figured out that additionalManifestEntries can be passed work workbox generateSW (use extendGenerateSWOptions in quasar.config.js under the  pwa object:
```
extendGenerateSWOptions(cfg) {
     cfg.additionalManifestEntries = [
     // additional entries here
  ]
}
```

**Reply by rezaser74** [reply]
hi
can you give me an example? 
i don't know how to do it
extendGenerateSWOptions(cfg) {
        console.log(cfg)
        cfg.additionalManifestEntries = [
        // additional entries here
          'offline.html',
        ]
      },

i did this and doesnt work

**Reply by Zaphod-Beeblebrox** [reply]
> hi can you give me an example? i don't know how to do it extendGenerateSWOptions(cfg) { console.log(cfg) cfg.additionalManifestEntries = [ // additional entries here 'offline.html', ] },
> 
> i did this and doesnt work

I would think HTML files would already be configured properly.  but make sure paths are correct too.  In my case, I needed to make sure AutoML (TensorFlow) Models were copied over .. my config looks like this:
   extendGenerateSWOptions(cfg) {
        cfg.additionalManifestEntries = [
          "models/group1-shard1of6.bin",
          "models/group1-shard2of6.bin",
          "models/group1-shard3of6.bin",
          "models/group1-shard4of6.bin",
          "models/group1-shard5of6.bin",
          "models/group1-shard6of6.bin",
        ];
and it successfully cached those bin files.

**Comment by felek000** [comment]
is there any function to remove folder/route/asset from list ?

**Comment by rezaser74** [comment]
thanks a lot for replying ☺️ i will try it
